ownCloud/Bookmarks/BookmarkViewController.swift

```diff
@@ -523,6 +534,18 @@ class BookmarkViewController: StaticTableViewController {
 		}
 	}
 
+	func completeAndDismiss(with hudCompletion: @escaping (((() -> Void)?) -> Void)) {
+		OnMainThread {
+			hudCompletion({
+				OnMainThread {
+					self.userActionCompletionHandler?(self.bookmark, true)
+					self.userActionCompletionHandler = nil
```

Comment:
This sending between threads and completion handlers carries the risk of `userActionCompletionHandler` being called more than once if the user if the user taps the Continue button in quick succession.

The previous version protected against this by putting `self.userActionCompletionHandler` into a variable, then setting it to `nil` - and only then sending off the block to the main thread.

Here's what I'd suggest:
```swift
	func completeAndDismiss(with hudCompletion: @escaping (((() -> Void)?) -> Void)) {
		guard let userActionCompletionHandler = self.userActionCompletionHandler else {
			return
		}

		self.userActionCompletionHandler = nil

		OnMainThread {
			hudCompletion({
				OnMainThread {
					userActionCompletionHandler(self.bookmark, true)
				}
				self.presentingViewController?.dismiss(animated: true, completion: nil)
			})
		}
	}
```